stdlib/LinearAlgebra/src/qr.jl

```diff
@@ -542,6 +542,36 @@ function rank(A::QRPivoted; atol::Real=0, rtol::Real=min(size(A)...) * eps(real(
     return something(findfirst(i -> abs(A.R[i,i]) <= tol, 1:m), m+1) - 1
 end
 
+function nullspace(A::QRPivoted; atol::Real=0, rtol::Real=min(size(A)...) * eps(real(float(one(eltype(A.Q))))) * iszero(atol))
+    m, n = size(A, 1), size(A, 2)
+    (m == 0 || n == 0) && return Matrix{eigtype(eltype(A.Q))}(I, n, n)
+
+    indstart = rank(A; atol, rtol) + 1
+    return A.Q[:, indstart:end]
+end
+
+function cond(A::QRPivoted, p::Real=2)
+    m = min(size(A.R)...)
+    if p == 2
+        maxv = abs(A.R[1,1])
+        return iszero(maxv) ? oftype(real(maxv), Inf) : maxv / abs(A.R[m,m])
+    elseif p == 1 || p == Inf
+        checksquare(A.R)
+        R = UpperTriangular(A.R)
+        try
+            Rinv = inv(R)
+            return opnorm(R, p)*opnorm(Rinv, p)
+        catch e
+            if isa(e, LAPACKException) || isa(e, SingularException)
+                return convert(float(real(eltype(A))), Inf)
+            else
+                rethrow()
+            end
+        end
+    end
+    throw(ArgumentError(lazy"p-norm must be 1, 2 or Inf, got $p"))
+end
```

Comment:
Since both `Q` and the permutation don't change norms, isn't this simply

```suggestion
cond(Q::QRPivoted, p::Real=2) = cond(Q.R, p)
```

? If we add this, we could also add the corresponding method for `Q::QR`.

Reply:
Indeed: https://github.com/JuliaLang/julia/issues/53214#issuecomment-1962035828.

Reply:
Q doesn't change 2-norms, but it changes 1 norms and infinity norms.

Reply:
Right. So, together with your comment below, I agree that we should forward this to `cond(Q.R, 2)` in the `p=2` case and otherwise throw and recommend using the original matrix, or, `cond(Matrix(Q), p)`.